object-browser-schema.js

```diff
@@ -0,0 +1,199 @@
+    },
+    pageSize: {
+      type: 'number',
+      default: 20,
+      description: 'Number of items fetched at a time'
+    },
+    initialPages: {
+      type: 'number',
+      default: 3,
+      description: 'Number of pages to load initially'
+    },
+    maxPages: {
+      type: 'number',
+      default: 3,
+      description: 'Number of pages to retain and keep updated. Set to 0 to to keep everything.'
+    },
+    filterBunsenModel: {
+      type: 'object',
+      description: 'Bunsen model for filter'
+    },
+    filterBunsenView: {
+      ref: 'https://github.com/ciena-blueplanet/bunsen-core/blob/master/src/validator/view-schemas/v2.js',
+      description: 'Bunsen view for filter'
+    },
+    sorting: {
+      type: 'array',
+      description: 'Available sort methods',
+      items: {
+        type: 'object',
+        additionalProperties: false,
+        properties: {
+          name: {
+            type: 'string',
+            description: 'Used both for display and configuration'
+          },
+          defaultDescending: {
+            type: 'boolean',
+            description: 'When selected the sort polarity will be changed to ascending or descending. Omit for normal behavior where ascending or descending state is retained.',
+          }
+        },
+        required: ['name']
+      }
+    },
+    currentSort: {
+      type: 'array',
+      description: 'Current sorting methods',
+      items: {
+        type: 'object',
+        additionalProperties: false,
+        properties: {
+          name: {
+            type: 'string',
+            description: 'Must match one of the names in the sorting list'
+          },
+          descending: {
+            type: 'boolean',
+            default: true
+          }
+        },
+        required: ['name']
+      }
+    },
+    selectionType: {
+      type: 'string',
+      description: 'Controls what type of selection is available',
+      enum: ['single', 'multi', 'none'],
+      default: 'multi'
+    },
+    selectedItems: {
+      type: 'array',
+      description: 'List of currently selected items',
+      items: {
+        description: 'Type should match the item id. Probably string.'
+      }
+    },
+    actionBarButtons: {
+      type: 'array',
+      description: 'Buttons in the action bar',
+      items: {
+        '$ref': '#/definitions/actionItem'
+      }
+    },
+    actionBarLinks: {
+      type: 'array',
+      description: 'Links in the action bar',
+      items: {
+        '$ref': '#/definitions/actionItem'
+      }
+    },
+    infoBar: {
+      type: 'object',
+      description: 'Info bar configuration',
+      properties: {
+        icon: {
+          type: 'object',
+          description: 'Info bar icon',
+          properties: {
+            name: {'string'},
+            pack: {'string'}
+          },
+          required: ['name']
+        },
+        title: {
+          type: 'string',
+        },
+        summary: {
+          type: 'string'
```

Comment:
We're shifting all of our components to support a JSON driven interface - my main comment here is there 'string' is likely to be insufficient (most of our summaries are not likely to be just text)